doc/manual.md

```diff
@@ -55,29 +53,27 @@ return values for functions.
 
 ### Records
 
-Records are nominal and should be declared in the top level, like the following
-example.
+Records types in Titan are nominal and should be declared in the top level,
+following the syntax `record Type (field: type)* end`. The following example
+declare a record `Point` with the fields `x` and `y` which are floats.
 
     record Point
         x: float
         y: float
     end
 
-    record Circle
-        center: Point
-        radius: float
-    end
-
-After the top level declaration, you may create records with the `.new`
-constructor:
-
-    local p = Point.new(1, 2)
-    local c = Circle.new(p, 3.5)
-
-You can access the fields with the dot operator:
-
-    local a = p.x
-    p.y = 2
+You can create records with initializer lists, assigning a value to each field
+either by it's name or positional order.
+For instance, you could initalize an instance of the record `Point` with: `{ x =
+3, y = 5 }`, `{ y = 5, x = 3 }` or `{ 3, 5 }`.
+In all those cases, the field `x` will receive the value `3` and the field `y`,
+`5`.
+Like arrays, Titan will try to infer the type of the initializer list, but it is
+necessary to specify the type of the record when declaring a new variable
+(`local p: Point = { 3, 5 }`).
+
```

Comment:
I don't know if this is out of scope for this PR but currently our type inference rules are undocumented and untested. I think we should rectify that if we are going to strongly rely on type inference for record instantiation.

Reply:
I don't know how to express that without being repetitive (there is already an explanation on the array section). Maybe we should have a section just for the inference rules, but that is indeed beyond the scope of this PR.